src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -397,15 +398,31 @@ BuildDistinctPlan(Query *masterQuery, Plan *subPlan)
 	bool distinctClausesHashable = true;
 	List *distinctClauseList = masterQuery->distinctClause;
 	List *targetList = copyObject(masterQuery->targetList);
-	List *columnList = pull_var_clause_default((Node *) targetList);
+	List *columnList = NIL;
 	ListCell *columnCell = NULL;
 	bool hasDistinctAggregate = false;
 
+
 	if (IsA(subPlan, Agg))
 	{
-		return subPlan;
+		/*
+		 * We don't need to add distinct plan if all of the columns used in group by
+		 * clause also used in distinct clause and we have an aggregation plan node.
```

Comment:
Isn't `we have an aggregation plan node.` equivalent of including `GROUP BY` clause? Reading the sentence makes me feel that we might have queries with `GROUP BY` which lacks `agg plan`